assets/aml-benchmark/components/src/dataset_downloader/main.py

```diff
@@ -0,0 +1,260 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Entry script for Dataset Downloader Component."""
+
+import argparse
+import os
+from typing import Optional, List
+from concurrent.futures import ProcessPoolExecutor
+
+from datasets import load_dataset, get_dataset_split_names, get_dataset_config_names
+import pandas as pd
+from azureml._common._error_definition.azureml_error import AzureMLError
+
+from utils.helper import get_logger, log_mlflow_params
+from utils.exceptions import (
+    swallow_all_exceptions,
+    BenchmarkValidationException,
+    DatasetDownloadException,
+)
+from utils.error_definitions import BenchmarkValidationError, DatasetDownloadError
+
+
+logger = get_logger(__name__)
+ALL = "all"
+
+
+def parse_args() -> argparse.Namespace:
+    """Parse command-line arguments."""
+    parser = argparse.ArgumentParser(description=f"{__file__}")
+    parser.add_argument(
+        "--dataset_name",
+        type=str,
+        default=None,
+        help="Name of the dataset to download.",
+    )
+    parser.add_argument(
+        "--configuration",
+        type=str,
+        default=None,
+        help="Configuration of the dataset to download.",
+    )
+    parser.add_argument(
+        "--split",
+        type=str,
+        default=None,
+        help="Split of the dataset to download.",
+    )
+    parser.add_argument(
+        "--url", type=str, default=None, help="URL of the dataset to download."
+    )
+    parser.add_argument(
+        "--output_dataset", type=str, required=True, help="Path to the dataset output."
+    )
+
+    args, _ = parser.parse_known_args()
+    logger.info(f"Arguments: {args}")
+    return args
+
+
+def resolve_configuration(
+    dataset_name: Optional[str], configuration: Optional[str]
+) -> List[Optional[str]]:
+    """
+    Get the list of configurations to download.
+
+    :param dataset_name: Name of the dataset to download.
+    :param configuration: Configuration of the dataset to download.
+    :return: list of configurations to download.
+    """
+    available_configs = get_dataset_config_names(dataset_name)
+
+    if configuration is None:
+        if len(available_configs) > 1:
+            mssg = (
+                f"Multiple configurations available for dataset '{dataset_name}'. Please specify either one of "
+                f"the following: {available_configs} or 'all'."
+            )
+            raise BenchmarkValidationException._with_error(
+                AzureMLError.create(BenchmarkValidationError, error_details=mssg)
+            )
+        else:
+            return [configuration]
```

Comment:
Would `available_configs` ever be empty?

Reply:
`available_configs` won't be empty. For datasets that do not have configuration listed on the HuggingFace UI, they return a random config name. We do not want to use it, configuration should be `None` in this case to download the dataset.